src/java.base/share/classes/java/lang/String.java

```diff
@@ -696,6 +696,13 @@ private String(Charset charset, byte[] bytes, int offset, int length) {
         }
     }
 
+    static String newStringLatin1NoRepl(byte[] bytes) {
```

Comment:
newStringLatin1NoRepl does not significantly help performance, but it simplifies the code. 

Of course, there is another advantage: because the call will be simpler, making it easier for the caller method to implement codeSize smaller than the default value of FreqInlineSize 325, and there are more opportunities to be inlined.

Reply:
Notice there is patch https://github.com/openjdk/jdk/pull/14655 (bug 8310901) for converting these usages to a new `newStringLatin1NoRepl`.

Reply:
Going from `jla.newStringNoRepl(bytes, ISO_...)` to `jla.newStringLatin1NoRepl(bytes)` isn't much of a code simplification, so this distracts a bit from the bulk of the changes in this PR. I agree that it *might* help inlining since the former calls into a large method, but I think we ought to focus on one thing at a time so if this doesn't significantly help on the benchmarks we have available it would be better to leave it out for now.

Reply:
I'll have a look at #14655 though I note that @RogerRiggs has already commented that this might not carry its own weight. These internal API bridges are a nuisance so we need to take care to make sure that any addition is actually necessary, not just convenient for the time being.

Reply:
Can you help me create an issue for jla.newStringNoRepl?

Reply:
I've recommended before to not add this convenience method, there's quite enough stuff in shared secrets and adds function that is too close to one already there.

Reply:
newStringLatin1NoRepl has been removed